pkg/rules/rules.go

```diff
@@ -167,6 +167,42 @@ type ResolveInput struct {
 	Headers        http.Header                   `json:"headers"`
 }
 
+func (r ResolveInput) ToKeyValues() []any {
```

Comment:
feels like there should be some library to use for this? not blocking though

Reply:
Yeah definitely considered. Probably would have to use reflection, being in the critical path, and only needing a subset of all the fields, I'd say this is fine